betty/core.py

```diff
@@ -0,0 +1,62 @@
+"""
+Provide tools to build core application components.
+"""
+
+from contextlib import AsyncExitStack
+from types import TracebackType
+from typing import Self, Any
+
+from betty.typing import internal
+
+
+@internal
+class CoreComponent:
```

Comment:
Document that this class is internal but it's public methods are, well, public. Maybe use a decorator for that too? Can the decorators add messages to the docstring?